packages/web-frontend/src/selectors/dashboardSelectors.js

```diff
@@ -58,12 +59,43 @@ export const selectCurrentInfoViewKey = createSelector(
 );
 
 export const selectCurrentExpandedViewContent = createSelector(
+  [selectCurrentInfoViewKey, state => state.dashboard.viewResponses],
+  (infoViewKey, viewResponses) => {
+    return viewResponses[infoViewKey];
+  },
+);
+
+export const selectIsEnlargedDialogVisible = createSelector(
   [
-    state => state.enlargedDialog.viewContent,
-    selectCurrentInfoViewKey,
-    state => state.dashboard.viewResponses,
+    selectCurrentDashboardGroupIdForExpandedReport,
+    selectCurrentOrgUnitCode,
+    selectCurrentExpandedViewId,
   ],
-  (viewContent, infoViewKey, viewResponses) => {
-    return viewContent || viewResponses[infoViewKey];
+  (dashboardGroupId, organisationUnitCode, viewId) =>
+    !!(dashboardGroupId && organisationUnitCode && viewId),
+);
+
+export const selectShouldUseDashboardData = createSelector(
+  [selectCurrentInfoViewKey, selectCurrentExpandedViewContent, (_, options) => options],
+  (candidateInfoViewKey, candidateViewContent, options) => {
+    const { startDate, endDate, infoViewKey, drillDownLevel } = options;
+
+    if (drillDownLevel > 0) return false;
+    if (candidateInfoViewKey !== infoViewKey) return false;
+    if (!candidateViewContent || candidateViewContent.type === 'matrix') return false;
+
+    const { startDate: defaultStartDate, endDate: defaultEndDate } = getDefaultDates(
```

Comment:
If they open the dashboard group on one day, leave it overnight and open a report the next day, it will refetch the data (perhaps) unnecessarily. But I don't think that's a huge issue

Reply:
I think that is actually a good thing.